src/classes/CRLP_Batch_Base.cls

```diff
@@ -134,93 +134,16 @@ public abstract class CRLP_Batch_Base {
         }
         this.recordIds = recordIds;
 
-        this.summaryObjectType = CRLP_RollupBatch_SVC.getSummaryObjectType(this.jobType);
+        this.summaryObjectType = CRLP_Rollup_SVC.getSummaryObjectType(this.jobType);
+
         this.statefulCacheOfRollupsToBeProcessed = CRLP_Rollup_SEL.getRollupDefinitionsToUse(this.jobType, this.jobFilter);
         this.isScheduledJob = System.isScheduled();
         this.customizableRollupSettings = UTIL_CustomSettingsFacade.getCustomizableRollupSettings();
 
-        switch on (jobType) {
-            when GAU {
-                this.detailObjectType = Allocation__c.SObjectType;
-                this.keyField = SObjectType.Allocation__c.fields.General_Accounting_Unit__c.getName();
-            }
-            when AccountHardCredit {
-                this.detailObjectType = Opportunity.SObjectType;
-                this.keyField = SObjectType.Opportunity.fields.AccountId.getName();
-            }
-            when AccountSoftCredit {
-                this.detailObjectType = Account_Soft_Credit__c.SObjectType;
-                this.keyField = SObjectType.Account_Soft_Credit__c.fields.Account__c.getName();
-            }
-            when ContactHardCredit {
-                this.detailObjectType = Opportunity.SObjectType;
-                this.keyField = SObjectType.Opportunity.fields.Primary_Contact__c.getName();
-            }
-            when ContactSoftCredit {
-                this.detailObjectType = OpportunityContactRole.SObjectType;
-                this.keyField = SObjectType.OpportunityContactRole.fields.ContactId.getName();
-            }
-            when AccountContactSoftCredit {
-                this.detailObjectType = OpportunityContactRole.SObjectType;
-                this.keyField = 'Contact.AccountId';
-            }
-            when RecurringDonations {
-                this.detailObjectType = Opportunity.SObjectType;
-                this.keyField = SObjectType.Opportunity.fields.npe03__Recurring_Donation__c.getName();
-            }
-        }
+        this.detailObjectType = CRLP_Rollup_SVC.getDetailObjectType(jobType);
+        this.keyField = CRLP_Rollup_SVC.getParentRelationshipFieldForQueryReference(jobType);
```

Comment:
There are two of these that return slightly different things:
* `getParentRelationshipFieldForQueryReference`
* `getParentRelationshipFieldForProcessingJob`

The only real difference is the field removed for Contact and AccountContact Soft Credits. This is because in these two rollup operations, the detail object changes from OCR (the object queried) to Partial Soft Credit (the object processed). I was trying to find a way to name these so it was clear what was being returned based on intent - "ForQuery" versus "ForProcessing"

I could try something like if think that's clearer:
* `getParentFieldNameForQuery`
* `getParentFieldNameForProcessing`